pkg/ensurance/executor/waterline_test.go

```diff
@@ -0,0 +1,46 @@
+package executor
+
+import (
+	"container/heap"
+	"strconv"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func (w WaterLine) verify(t *testing.T, i int) {
+	t.Helper()
+	n := w.Len()
+	j1 := 2*i + 1
+	j2 := 2*i + 2
+	if j1 < n {
+		if w.Less(j1, i) {
+			t.Errorf("heap invariant invalidated [%d] = %d > [%d] = %d", i, w[i].Value(), j1, w[j1].Value())
+			return
+		}
+		w.verify(t, j1)
+	}
+	if j2 < n {
+		if w.Less(j2, i) {
+			t.Errorf("heap invariant invalidated [%d] = %d > [%d] = %d", i, w[i].Value(), j1, w[j2].Value())
+			return
+		}
+		w.verify(t, j2)
+	}
+}
+
+func TestSort(t *testing.T) {
+	h := WaterLine{}
+
+	for i := 20; i > 0; i-- {
```

Comment:
What the effect of this test function, can we have a more general test case for the function?